lib/trufi_models.dart

```diff
@@ -443,20 +442,44 @@ class PlanItineraryLeg {
   }
 
   String _carTypeString(TrufiLocalization localization) {
-    String carType = routeLongName?.toLowerCase() ?? "";
-    return mode == 'CAR'
-        ? localization.instructionVehicleCar()
-        : carType.contains(type_trufi)
-            ? localization.instructionVehicleTrufi()
-            : carType.contains(type_micro)
-                ? localization.instructionVehicleMicro()
-                : carType.contains(type_minibus)
-                    ? localization.instructionVehicleMinibus()
-                    : carType.contains(type_gondola)
-                        ? localization.instructionVehicleGondola()
-                        : carType.contains(type_light_rail)
-                            ? localization.instructionVehicleLightRail()
-                            : localization.instructionVehicleBus();
+    switch (mode) {
```

Comment:
It would be amazing in the future if we can somehow create an Enum out of the different strings. I took a brief look but I believe we first need some kind of Repo Layer with a good mapper.

And something like a JSON Serializer

Reply:
Some of these types are also very specific to the country/region the app is used in. We should probably introduce a way to extend the list from the host app and move things like Microbus/Trufi to it. The core would still know how to translate generic trains, trams, buses etc.